### ISSUE TITLE
Trailing comma in function literal is incorrectly formatted and results in false positive at next invocation

### ISSUE THREAD

**Issue by paul-dingemans** [pinterest/ktlint#1911]
Given code below and that trailing comma's on declaration site are enabled:
```
val foo = {
        string: String,
        int: Int ->
    // do something
}
```
then this code is formatted as:
```
val foo = {
        string: String,
        int: Int, ->
    // do something
}
```
and on the next invocation of ktlint, it reports violation below:
```
Whitespace after ',' is missing (experimental:parameter-list-spacing)
```

On the first invocation, the code should be formatted as:
```
val foo = {
        string: String,
        int: Int,
    ->
    // do something
}
```
and no violation may be reported on following invocations
